scene/main/node.cpp

```diff
@@ -1346,6 +1346,46 @@ void Node::remove_child(Node *p_child) {
 	remove_child_notify(p_child);
 	p_child->notification(NOTIFICATION_UNPARENTED);
 
+	// Note that calling _set_tree() could have caused
+	// another node to be deleted from the same parent,
+	// the childlist could be different, and thus
+	// idx by this point is not guaranteed to be correct.
+	// We therefore recalculate idx here, for safety, if it has changed.
+
+	// Child count could have changed.
+	child_count = data.children.size();
+
+	if ((p_child->data.pos != idx) || (idx >= child_count) || (data.children[idx] != p_child)) {
```

Comment:
I didn't write a comment here, but just to make it clear for review, there are three possibilities here (for detecting a change to `p_child`'s index):
* The child itself _knows_ that its index has changed
* The previous index is now outside the range of the new child count
* The child of the parent given by the index is no longer `p_child` (this can be checked providing `idx < child_count` in the previous)